db/write_stall_stats.cc

```diff
@@ -137,17 +143,21 @@ std::string WriteStallStatsMapKeys::CauseConditionCount(
     WriteStallCause cause, WriteStallCondition condition) {
   std::string cause_condition_count_name;
 
-  std::string cause_name;
-  if (isCFScopeWriteStallCause(cause) || isDBScopeWriteStallCause(cause)) {
+  std::string cause_name, condition_name;
+  if ((isCFScopeWriteStallCause(cause) &&
+       InternalCFStat(cause, condition) !=
+           InternalStats::INTERNAL_CF_STATS_ENUM_MAX) ||
+      (isDBScopeWriteStallCause(cause) &&
+       InternalDBStat(cause, condition) != InternalStats::kIntStatsNumMax)) {
+    assert(cause < WriteStallCause::kNone &&
+           condition < WriteStallCondition::kNormal);
     cause_name = kWriteStallCauseToHyphenString[static_cast<uint32_t>(cause)];
+    condition_name =
+        kWriteStallConditionToHyphenString[static_cast<uint32_t>(condition)];
   } else {
-    assert(false);
-    return "";
```

Comment:
Seems not a bad approach. Should we still `assert(false)` or return a `Status::InvalidArgument` if it must be called in such a way? It seems easy to accidentally query `db_values[key]` when `key` is `InvalidMapKey()` and get zero.

Reply:
Yeah we can do assert(false) before `return InvalidMapKey();`

Reply:
fixed

Reply:
Sorry I meant the old code looked fine to me. To evolve that, I'd either go with a `Status` returning function or a validity checking function. I'm not sure about adding a special string to the public API to denote programming errors. It might be hard to use since it actually is a valid key in the map of strings.

Reply:
Oh got you - fixed it by making `CauseConditionCount()` a "Status returning function" - will update internal callers soon